lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/config/LndConfig.scala

```diff
@@ -0,0 +1,268 @@
+    val linesToSearchIn =
+      lines.takeWhile(l => !l.trim.startsWith("[") && !l.trim.startsWith(";"))
+    val collect = collectFrom(linesToSearchIn)(_)
+    collect { case (`key`, value) =>
+      value
+    }.headOption
+  }
+
+  /** Converts a string to an InetSocketAddress */
+  private def toInetSocketAddress(string: String): InetSocketAddress = {
+    val uri = new URI(string)
+    new InetSocketAddress(uri.getHost, uri.getPort)
+  }
+
+  lazy val bitcoindUser: String = getValue("bitcoind.rpcuser").getOrElse("foo")
+  lazy val bitcoindPass: String = getValue("bitcoind.rpcpass").getOrElse("bar")
+
+  lazy val listenBinding: URI = new URI({
+    val baseUrl = getValue("listen").getOrElse("127.0.0.1:9735")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+  })
+
+  lazy val rpcBinding: URI = new URI({
+    val baseUrl = getValue("rpclisten").getOrElse("127.0.0.1:10009")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+  })
+
+  lazy val restBinding: URI = new URI({
+    val baseUrl = getValue("restlisten").getOrElse("127.0.0.1:8080")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+  })
+
+  lazy val bitcoindBinding: URI = new URI({
+    val baseUrl =
+      getValue("bitcoind.rpchost").getOrElse(s"127.0.0.1:${network.rpcPort}")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+  })
+
+  lazy val zmqpubrawblock: Option[InetSocketAddress] =
+    getValue("bitcoind.zmqpubrawblock").map(toInetSocketAddress)
+
+  lazy val zmqpubrawtx: Option[InetSocketAddress] =
+    getValue("bitcoind.zmqpubrawtx").map(toInetSocketAddress)
+
+  lazy val debuglevel: LogLevel = getValue("debuglevel")
+    .flatMap(LogLevel.fromStringOpt)
+    .getOrElse(LogLevel.Debug)
+
+  /** Creates a new config with the given keys and values appended */
+  def withOption(key: String, value: String): LndConfig = {
+    val ourLines = this.lines
+    val newLine = s"$key=$value"
+    val lines = newLine +: ourLines
+    val newConfig = LndConfig(lines, datadir)
+    logger.debug(
+      s"Appending new config with $key=$value to datadir=${datadir.getAbsolutePath}")
+    LndConfig.writeConfigToFile(newConfig, datadir)
+
+    newConfig
+  }
+
+  def withDatadir(newDatadir: File): LndConfig = {
+    LndConfig(lines, newDatadir)
+  }
+
+  lazy val lndInstance: LndInstance = LndInstance(
+    datadir.toPath,
+    network,
+    listenBinding,
+    restBinding,
+    rpcBinding,
+    PasswordBased(bitcoindUser, bitcoindPass),
+    bitcoindBinding,
+    ZmqConfig(rawBlock = zmqpubrawblock, rawTx = zmqpubrawtx),
+    debuglevel
+  )
+}
+
+object LndConfig extends Logging {
+
+  /** The empty `lnd` config */
+  lazy val empty: LndConfig = LndConfig("", DEFAULT_DATADIR)
+
+  /** Constructs a `lnd` config from the given string,
+    * by splitting it on newlines
+    */
+  def apply(config: String, datadir: File): LndConfig =
```

Comment:
Can we get some sort of common interface across all this? I believe this is the same with bitcoind/eclair too 

https://github.com/bitcoin-s/bitcoin-s/blob/f593cf063e53b2c8e6b867d5c975d824694b31cd/bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindInstance.scala#L123

Reply:
This is sort of similar, but is meant for `AppConfig` https://github.com/bitcoin-s/bitcoin-s/blob/ee7c96245e99325d1386464c4462d36dce7bcafb/db-commons/src/main/scala/org/bitcoins/db/AppConfigFactory.scala#L9